slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -194,6 +194,17 @@ ::grpc::Status SLAMServiceImpl::GetPointCloudMap(const GetMapRequest *request,
     }
 }
 
+::grpc::Status SLAMServiceImpl::GetInternalState(
+    ServerContext *context, const GetInternalStateRequest *request,
+    GetInternalStateResponse *response) {
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        std::string buf = map_builder.SaveMapToStream(path_to_map);
+        response->set_internal_state(buf);
```

Comment:
Is there any concern that `buf` won't fit in the response?

Reply:
Currently, the large map of the office produces 4.7 MB of data in the pbstream format. GRPC protobuf has a hard limit of 2 GB for an individual message (although this can be increased using protobuf-c as we do in RDK). Issues of RAM can appear depending on the processor at as low as 8-10 MB as the whole message needs to be stored in RAM during the transfer, however our pis have 8GB (some of which will be in use for other things) so we should be fine for a while. 

I think we are fine for now but as we discussed for PCDs a chunking format may need to be investigated in the future

Reply:
sounds good, thank you for checking this!